### PR TITLE
add `txpool_inspect`

### PR DESCRIPTION
**Pull request by tcoratger**
Time spent on this PR:

Resolves: #<Issue number>

# Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing

# What is the new behavior?

# Does this introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: src/eth_provider/provider.rs [hunk 1]
```diff
@@ -128,6 +128,8 @@ pub trait EthereumProvider {
         &self,
         block_id: Option<BlockId>,
     ) -> EthProviderResult<Option<Vec<reth_rpc_types::Transaction>>>;
+    /// Returns a vec of pending pool transactions.
+    async fn txpool_transactions(&self) -> EthProviderResult<Vec<StoredPendingTransaction>>;
```
Comment (greged93) [inline src/eth_provider/provider.rs:132]: I'd prefer we avoid introducing our types in the trait, keep it reth only

File: tests/tests/txpool_api.rs [hunk 2]
```diff
@@ -199,3 +199,83 @@ async fn test_txpool_status(#[future] katana: Katana, _setup: ()) {
...
+    // Start the Kakarot RPC server and retrieve the server address and handle
+    let (server_addr, server_handle) =
+        start_kakarot_rpc_server(&katana).await.expect("Error setting up Kakarot RPC server");
+
+    // Generate a vector of random bytes
+    let bytes: Vec<u8> = (0..RANDOM_BYTES_SIZE).map(|_| rand::random()).collect();
+    let mut unstructured = arbitrary::Unstructured::new(&bytes);
+
+    // Generate 10 pending transactions and add them to the database
+    let mut pending_transactions = Vec::new();
```
Comment (greged93) [inline tests/tests/txpool_api.rs:216]: I think we have a lot of boiler plate in these tests no? Could we use a few helpers maybe?

File: src/eth_provider/provider.rs (hunk 1 above)
Comment (tcoratger) [reply src/eth_provider/provider.rs]: Agreed, just fixed

File: tests/tests/txpool_api.rs (hunk 2 above)
Comment (tcoratger) [reply tests/tests/txpool_api.rs:216]: It's true but I told myself that:
- It was more flexible knowing that depending on the bugs that we encounter in the future, these testing conditions may evolve and not be the same for all unit tests.
- It was rather readable and compartmentalized like that.
- As we already have quite a few utils for testing, I didn't want to add too much.

Maybe we can do that on a follow up good first issue with a smart arrangement if necessary?

Comment (greged93) [reply tests/tests/txpool_api.rs:216]: Sure sounds good, I was just maybe thinking a simple function (doesn't even have to be in the test_utils) that returns pending transactions for example. But good first issue works!

**Review by greged93 — APPROVED** [review]
can you just write the issue and then merge!

**Comment by tcoratger** [comment]
> can you just write the issue and then merge!

Done https://github.com/kkrt-labs/kakarot-rpc/issues/1108

**Pull request merged**